packages/web-app-files/tests/unit/components/Upload/DetailsWidget.spec.js

```diff
@@ -0,0 +1,77 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import DesignSystem from 'owncloud-design-system'
+
+import DetailsWidget from '@files/src/components/Upload/DetailsWidget.vue'
+
+const localVue = createLocalVue()
+localVue.use(DesignSystem)
+
+const selectors = {
+  oCProgress: 'oc-progress-stub',
+  itemName: '#item-name',
+  itemSize: '#item-size',
+  progressText: '.oc-progress-text'
+}
+
+const items = {
+  id: 1,
+  name: 'lorem.txt',
+  size: '100',
+  progress: 20
+}
+
+describe('DetailsWidget component', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('prop "items" is required', () => {
+    expect(DetailsWidget.props.items.required).toBe(true)
```

Comment:
cool, in some other places and in ODS I've inspected the console error to check if the prop is required or not. imo checking the console is not a good way to test it. should we change those back like this?

Reply:
CC @individual-it

Reply:
let's change that but in separate PR

Reply:
We'll prepare some feedback on the unit tests early next week :) but I think checking for props that are required counts as "testing vendor code" which we should try to avoid

Reply:
Seeing this now. Agreed, a unit test should not test code of other units.